eth/downloader/downloader.go

```diff
@@ -573,6 +595,13 @@ func (d *Downloader) syncWithPeer(p *peerConnection, hash common.Hash, td, ttd *
 				return err
 			}
 		}
+
+		// if enable history segment, force reset freezer tail
+		if d.prevSegment != nil && localHeight == 0 {
+			if err := d.blockchain.FreezerDBReset(origin, origin); err != nil {
```

Comment:
this function only called here, and tail == head,

so much code on path ` tail !=head` will never be called?

if so, we can just simplify the logic